chore: improve code readability and style in regressions.py and splitSources.py

Pronoss opened the pull request:
- Updated docstrings and comments for clarity and grammatical correctness.
- Fixed phrasing: "sources into their own files" → "sources into separate files".
- Improved uncaught exception handling print formatting in `splitSources.py`.
- Replaced manual string slicing with `startswith()` for better readability and Python best practices.
- Standardized class naming: `regressor` → `Regressor`.
- Improved comment wording for non-ASCII handling.
- Minor whitespace and indentation adjustments.

Pronoss committed:
chore(scripts): improve readability of regressions.py and splitSources.py

Co-authored-by: clonker <1685266+clonker@users.noreply.github.com>

Pronoss commented:
Hi @solidity – just a friendly reminder about PR #15959 (“chore: improve code readability and style in regressions.py and splitSources.py”). It updates docstrings, switches string slicing to `startswith()`, standardizes the `Regressor` class name, and refines exception handling in `splitSources.py`. When you get a moment, could you take a look? Thanks! 🚀

clonker commented on scripts/splitSources.py:
```diff
@@ -18,7 +18,7 @@ def uncaught_exception_hook(exc_type, exc_value, exc_traceback):
     # as a critical error (because of the uncaught exception) and will
     # terminate further execution.
-    print("Unhandled exception: %s", "".join(traceback.format_exception(exc_type, exc_value, exc_traceback)))
+    print("Unhandled exception:" + "".join(traceback.format_exception(exc_type, exc_value, exc_traceback)))
```
```suggestion
    print("Unhandled exception: " + "".join(traceback.format_exception(exc_type, exc_value, exc_traceback)))
```
Otherwise this will put the trace directly next to the `:`. :)

Pronoss commented:
Thank you, @clonker, for catching and fixing that typo! Everything looks good to me now.

clonker commented:
Could you please squash into one or two commits and rebase? We don't want to have merge commits on develop.

Pronoss commented:
Hi @clonker, 

I’ve squashed all changes into a single commit and rebased onto `develop` as requested.
Let me know if there’s anything else to adjust—thanks!

clonker reviewed (approved):
Looks good now, thanks @Pronoss!
